IoHandle.md

```diff
@@ -0,0 +1,387 @@
+    virtual T addressFromString(const std::string& address, uint32_t port=0) PURE;
+    virtual const std::string& addressToString() PURE;
+    virtual const std::string& addressAsString() const PURE;
+
+    virtual bool isAnyAddress() const PURE;
+    virtual bool isUnicastAddress() const PURE;
+
+    virtual IpVersion version() const PURE;
+
+    operator(T) const { return(address_); }
+    operator(std::string) const { return(asString()); }
+
+   bool	operator == (const T& lhs) {
+      return (address_.addressBytes() == lhs.addressBytes()); } 
+  bool    operator != (const T& lhs) {
+      return address_.addressBytes() !=  lhs.addressBytes()); }
+ 
+protected:
+
+    T    		 	          address_;
+    std::string    	 	friendly_address_;
+    std::string    	 	friendly_name_;
+    const IoAddressType 	type_;
+};
+
+// Implementations for IoAddressType. The details for Ipv4 sockets are given
+// below, and the v6 socket address implementation is similar. The IoAddressPipe
+// requires more thought still and is in progress.
+
+class IoSockAddrIpv4 : public IoAddressBase<sockaddr_in, uint32_t> {
+public:
+   explicit IoSockAddrIpv4(const sockaddr_in& address) 
+: public IoAddressBase(address),type_(Ipv4) { addressToString(); }
+   explicit IoSockAddrIpv4(const sockaddr_in* address) 
+: public IoAddressBase(*address),type_(Ipv4) { addressToString(); }
+   explicit IoSockAddrIpv4(const std::string& address) 
+: public IoAddressBase(address),type_(Ipv4) 
+     { address_= addressFromString(address); }
+  IoSockAddrIpv4(const std::string& address, uint32_t port) 
+: public IoAddressBase(address),type_(Ipv4) 
+{ address_= addressFromString(address,port); }
+   explicit IoSockAddrIpv4(uint32_t port):type_(Ipv4)
+{ address_= addressFromString("",port); }
+
+   sockaddr_in address() const { return address_; }
+   uint32_t addressBytes() const { return address_.sin_addr.s_addr; }
+   uint32_t port() const override { return ntohs(address_.sin_port); }
+   bool isEmpty() const override { return addressBytes()==0 ; }
+   IoAddressType type() const override { return type_; }
+
+   // The following is unclear- this next method replaces socketFromSocketType() but
+   // maybe it should be somewhere else?
+   int socketFromIoType(IoType type) const;
+ 
+   const sockaddr_in&  addressFromString(const std::string& address,
+       uint32_t port=0) override;
+   const std::string& addressToString() override;
+   const std::string& addressAsString() const override { return friendly_address_;
+};
+
+```
+
+Note also that the methods addressFromFd() and peerAddressFromFd() should be methods in a derived class from a base class that has virtual methods addressFromIoHandle() and peerAddressFromIoHandle(), as opposed to addressFromSockAddr() which pertains only to sockets. 
+
+###### Transport Sockets
+
+The transport sockets will be redefined by the class IoTransport, with derivatives for sockets and for VPP.
```

Comment:
In this current PR, they weren't being replaced, just renamed to IoTransport. But this is irrelevant now, since the changes will be confined to ioHandle.